vllm/core/block_manager_v1.py

```diff
@@ -326,15 +327,19 @@ def allocate(self, seq_group: SequenceGroup) -> None:
         #
         # NOTE: Here we assume that all sequences in the group have the same
         # decoder prompt.
-        seq = seq_group.get_seqs(status=SequenceStatus.WAITING)[0]
+        wait_seqs = seq_group.get_seqs(status=SequenceStatus.WAITING)
+        seq = wait_seqs[0]
         block_table: BlockTable = \
             self._allocate_sequence(seq,
                                     seq_group.num_seqs(),
                                     is_encoder_decoder)
 
         # Assign the self-attention block tables for each sequence.
-        for seq in seq_group.get_seqs(status=SequenceStatus.WAITING):
-            self.block_tables[seq.seq_id] = block_table.copy()
+        if len(wait_seqs) == 1:
+            self.block_tables[wait_seqs[0].seq_id] = block_table
+        else:
+            for seq in seq_group.get_seqs(status=SequenceStatus.WAITING):
+                self.block_tables[seq.seq_id] = block_table.copy()
```

Comment:
```suggestion
            self.block_tables[seq.seq_id] = block_table
        else:
            for seq in wait_seqs:
                self.block_tables[seq.seq_id] = block_table.copy()
```

does this make sense?

Reply:
Addressed https://github.com/vllm-project/vllm/pull/7364